lib/Onyx.js

```diff
@@ -195,7 +195,12 @@ function keysChanged(collectionKey, collection) {
                     newData = lodashMerge(newData, prevState[subscriber.statePropertyName]);
                 }
                 return {
-                    [subscriber.statePropertyName]: newData,
+                    // If `newData` happens to not exist, then set the state back to the previous value because
+                    // if the state is set to an undefined value, then the item reverts to whatever is in
+                    // defaultProps
+                    [subscriber.statePropertyName]: !_.isUndefined(newData)
+                        ? newData
+                        : prevState[subscriber.statePropertyName],
```

Comment:
Thinking more on this... this could maybe be filed under "unrelated" or "not a problem", but I find it interesting (or rather incorrect/inefficient) that if a subscriber is connected to a single key, but it's data hasn't changed at all ... that we are still calling `setState()` with previous data?

That's not super great because it means that we will still trigger re-renders on those connected components and for no reason at all. So, probably if we have `undefined` then we should do less than set the previous state again and should do... well... nothing?

Reply:
Yeah, I think you make a good point. While it's not a problem currently, it's not a great design either. The reason I put it inside `setState` is because that's where it is getting `newData`. But now that I look at it closely, `newData` should be able to be retrieved before calling `setState()` and in the case where it is undefined, we can just skip calling `setState()`. How's that sound?

Reply:
Yep, sounds good to me. 👍